beacon-chain/rpc/prysm/v1alpha1/validator/proposer.go

```diff
@@ -38,6 +38,10 @@ func (vs *Server) GetBeaconBlock(ctx context.Context, req *ethpb.BlockRequest) (
 	ctx, span := trace.StartSpan(ctx, "ProposerServer.GetBeaconBlock")
 	defer span.End()
 	span.AddAttributes(trace.Int64Attribute("slot", int64(req.Slot)))
+
+	// update fork choice head before proposing
+	vs.HeadUpdater.UpdateAndSaveHead(ctx)
```

Comment:
I rather have us do it here: https://github.com/prysmaticlabs/prysm/blob/16bbf5602fc1b0756c4e2e61cfcb220a85de117b/beacon-chain/rpc/prysm/v1alpha1/validator/proposer_phase0.go#L84

As close to getting `parentRoot` the better.  We also don't want to update and save head if `vs.SyncChecker.Syncing()`  is `true`